ludwig/features/vector_feature.py

```diff
@@ -0,0 +1,445 @@
+            self,
+            regularizer,
+            dropout_rate,
+            is_training=False,
+            **kwargs
+    ):
+        placeholder = self._get_input_placeholder()
+        logger.debug('  placeholder: {0}'.format(placeholder))
+
+        feature_representation, feature_representation_size = self.encoder_obj(
+            placeholder,
+            self.vector_size,
+            regularizer,
+            dropout_rate,
+            is_training=is_training
+        )
+        logger.debug(
+            '  feature_representation: {0}'.format(feature_representation)
+        )
+
+        feature_representation = {
+            'name': self.name,
+            'type': self.type,
+            'representation': feature_representation,
+            'size': feature_representation_size,
+            'placeholder': placeholder
+        }
+        return feature_representation
+
+    @staticmethod
+    def update_model_definition_with_metadata(
+            input_feature,
+            feature_metadata,
+            *args,
+            **kwargs
+    ):
+        for key in ['vector_size']:
+            input_feature[key] = feature_metadata[key]
+
+    @staticmethod
+    def populate_defaults(input_feature):
+        set_default_value(input_feature, 'tied_weights', None)
+        set_default_value(input_feature, 'preprocessing', {})
+
+
+class VectorOutputFeature(VectorBaseFeature, OutputFeature):
+    def __init__(self, feature):
+        super().__init__(feature)
+        self.type = VECTOR
+        self.vector_size = 0
+
+        self.loss = {'type': MEAN_SQUARED_ERROR}
+
+        _ = self.overwrite_defaults(feature)
+
+        self.decoder = 'fc_stack'
+        feature['fc_size'] = self.vector_size
+        self.decoder_obj = self.get_vector_decoder(feature)
+
+    def get_vector_decoder(self, decoder_parameters):
+        return get_from_registry(
+            self.decoder, vector_decoder_registry)(
+            **decoder_parameters
+        )
+
+    def _get_output_placeholder(self):
+        return tf.placeholder(
+            tf.float32,
+            [None, self.vector_size],
+            name='{}_placeholder'.format(self.name)
+        )
+
+    def _get_measures(self, targets, predictions):
+
+        with tf.variable_scope('measures_{}'.format(self.name)):
+            error_val = get_error(
+                targets,
+                predictions,
+                self.name
+            )
+
+            absolute_error_val = tf.reduce_sum(
+                get_absolute_error(targets, predictions, self.name), axis=1
+            )
+
+            squared_error_val = tf.reduce_sum(
+                get_squared_error(targets, predictions, self.name), axis=1
+            )
+
+            # TODO - not sure if this is correct
+            r2_val = tf.reduce_sum(get_r2(targets, predictions, self.name))
+
+        return error_val, squared_error_val, absolute_error_val, r2_val
+
+    def vector_loss(self, error_val, squared_error_val):
```

Comment:
other losses may not use the errors but do something different, so it's better to have gt and predictions here

Reply:
Got it. I wanted to avoid the repetitive computations.

Reply:
Makes sense, but itdeally TF will optimize the computation graph for that.